lib/plugins/aws/deploy/compile/events/schedule/index.js

```diff
@@ -54,19 +76,28 @@ class AwsCompileScheduledEvents {
 
             const normalizedFunctionName = functionName[0].toUpperCase() + functionName.substr(1);
 
-            const scheduleTemplate = `
+            let scheduleTemplate = [`
               {
                 "Type": "AWS::Events::Rule",
                 "Properties": {
                   "ScheduleExpression": "${ScheduleExpression}",
                   "State": "${State}",
                   "Targets": [{
                     "Arn": { "Fn::GetAtt": ["${normalizedFunctionName}LambdaFunction", "Arn"] },
-                    "Id": "${functionName}Schedule"
+                    "Id": "${functionName}Schedule"`,
+            ];
+            if (Input) {
+              scheduleTemplate.push(`,"Input": "${Input}"`);
+            }
+            if (InputPath) {
+              scheduleTemplate.push(`,"InputPath": "${InputPath}"`);
+            }
+            scheduleTemplate.push(`
                   }]
                 }
               }
-            `;
+            `);
+            scheduleTemplate = scheduleTemplate.join('');
```

Comment:
Just tested in `v4.3.2` and it works. Furthermore you should be able to put (almost) any JS statement inside the `${}` blocks